repo2docker/utils.py

```diff
@@ -69,6 +70,22 @@ def chdir(path):
     finally:
         os.chdir(old_dir)
 
+@contextmanager
+def open_utf8convert_read(path):
```

Comment:
I'm not a huuuge fan of the name as it is long and still I feel like I have to go read its code to know what it does.

Some ideas, neither of which I really like either: `open_with_chardet`, `open_guess_encoding`, `guess_encoding`, `open_auto_encoding` :-/

Reply:
Sorry, but I'm not that good at naming things. I changed the name 3 times before coming up with ``open_utf8convert_read`` (the other ones were waaay worse :disappointed_relieved:).

I really like ``open_guess_encoding``, thanks :heart:

Reply:
No worries! Finding good names (not too long, tell you what the thing is, etc, etc) is one of the hardest things. Also totally worth doing I think because future us will thank past us for giving the thing a good name :)